Regression: Action can no longer be used for non-PR triggers

#### Describe the bug

If the action is executed in a job that doesn't have an attached PR (e.g. one that's triggered on GitHub releases), then the action will fail with 

```
...
Run # Post output.
gh: Not Found (HTTP 404)
Error: Process completed with exit code 1.
```

This is not particularly descriptive error (it doesn't say what it tried to do when it got a 404), but I _believe_ it's caused by this line being executed:

<table><thead><tr><th>Code executed</th></tr></thead><tbody><tr><td>

```
  # Check if the PR contains a bot comment with the same identifier.
  list_comments=$(gh api /repos/foobar/my-repo/issues/0/comments --header "$GH_API" --method GET --paginate)
```

</td></tr></tbody><thead><tr><th>Source code</th></tr></thead><tbody><tr><td>

https://github.com/OP5dev/TF-via-PR/blob/5b5ec65297fd2f686f018a0490ac059dd543b157/action.yml#L479-L480

</td></tr></tbody></table>

As you can see, it's trying to fetch comments for issue "0", which is likely where the 404 error comes from.

I believe this problem was introduced in https://github.com/OP5dev/TF-via-PR/pull/503, since it moved the `list_comments=...` line out of the `if [[ "$create_comment" == "true" && "${{ steps.identifier.outputs.pr }}" != "0" ]]; then` gate that it used to be behind.

#### To Reproduce

Run a workflow triggered on e.g.

```
on:
  release:
    types: [published]
```

Call the action there.

#### Expected behavior

The action should not attempt to read comments from issue 0.

#### Screenshots

<img width="1316" height="101" alt="Image" src="https://github.com/user-attachments/assets/3d351d52-e4a3-4755-97fa-db4ce0af12f5" />

Working with @birjj I want to add some context, to maybe make it easier to understand what we are trying to do, and why rolling back is not an option.

---

### What are we trying to do?

We have built a Reusable Workflow utilizing this Action, and we want to use it as demonstrated below:

![](https://github.com/tv2/dvt-docs/blob/main/docs/iac/approaches/tf-via-github-actions.png)

Everything is working smoothly except the deployment from GitHub Release events.

The workflow that is failing is the below one:

```yaml
name: Terraform Apply (Stage)

on:
  release:
    types: [published]

jobs:
  terraform-plan-stage:
    name: "Stage Plan"
    uses: XXX
    with:
      aws-role-arn: "XXX"
      aws-region: "XXX"
      working-directory: "./tf"
      tool: "terraform"
      tool-version: "latest"
      arg-backend-config: "../../vars/stage-backend.hcl"
      arg-var-file: "../../vars/stage.tfvars"
      comment-on-pr: "never"
    secrets:
      encryption-key: ${{ secrets.TF_PASSPHRASE }}

  terraform-apply-stage:
    name: "Stage"
    uses: XXX
    needs: terraform-plan-stage
    with:
      aws-role-arn: "XXX"
      aws-region: "XXX"
      working-directory: "./tf"
      tool: "terraform"
      tool-version: "latest"
      arg-backend-config: "../../vars/stage-backend.hcl"
      arg-var-file: "../../vars/stage.tfvars"
      gh-environment: "stage"
      comment-on-pr: "never"
    secrets:
      encryption-key: ${{ secrets.TF_PASSPHRASE }}
```

Which uses the following `op5dev/tf-via-pr` jobs in the background (via its Reusable Workflow)

```yaml
# For running terraform plan
on:
  workflow_call:
jobs:
  terraform-plan:
    ...
    runs-on: ubuntu-latest

    permissions:
      actions: read # Required to identify workflow run.
      checks: write # Required to add status summary.
      contents: read # Required to checkout repository.
      pull-requests: write # Required to add PR comment.
      id-token: write # Required to assume role.

    strategy:
      matrix:
        module: ${{ fromJson(needs.auto-discover-modules.outputs.modules) }}

    steps:
	    ...
		- name: Run Terraform Plan
		  # TODO: Update to 13.7.0 when a fix for retrieving comments has been made conditional
		  uses: op5dev/tf-via-pr@8a5419f58e07d28721f4f9c907c85f3c5648c2b1 # v13.6.1
		  with:
		    working-directory: ${{ inputs.working-directory }}/${{ matrix.module }}
		    command: "plan"
		    arg-refresh: "true"
		    arg-lock-timeout: ${{ inputs.arg-lock-timeout }}
		    arg-backend-config: ${{ inputs.arg-backend-config }}
		    arg-var-file: ${{ inputs.arg-var-file }}
		    arg-workspace: ${{ inputs.arg-workspace }}
		    plan-encrypt: ${{ secrets.encryption-key }}
		    plan-parity: true # Prevents stale apply within merge queue.
		    tool: ${{ inputs.tool }}
		    comment-pr: ${{ inputs.comment-on-pr }}
```

```yaml
jobs:
  terraform-apply:
    ...
    runs-on: ubuntu-latest

    permissions:
      actions: read # Required to identify workflow run.
      checks: write # Required to add status summary.
      contents: read # Required to checkout repository.
      pull-requests: write # Required to add PR comment.
      id-token: write # Required to assume role.

    strategy:
      matrix:
        module: ${{ fromJson(needs.auto-discover-modules.outputs.modules) }}
      max-parallel: 1 # runs applies sequentially
    
    steps:
        ....
		# For running terraform apply
		- name: Run Terraform Apply
		  # TODO: Update to 13.7.0 when a fix for retrieving comments has been made conditional
		  uses: op5dev/tf-via-pr@8a5419f58e07d28721f4f9c907c85f3c5648c2b1 # v13.6.1
		  with:
		    working-directory: ${{ inputs.working-directory }}/${{ matrix.module }}
		    command: "apply"
		    arg-refresh: "false"
		    arg-lock-timeout: ${{ inputs.arg-lock-timeout }}
		    arg-backend-config: ${{ inputs.arg-backend-config }}
		    arg-var-file: ${{ inputs.arg-var-file }}
		    arg-workspace: ${{ inputs.arg-workspace }}
		    plan-encrypt: ${{ secrets.encryption-key }}
		    plan-parity: true # Prevents stale apply within merge queue.
		    tool: ${{ inputs.tool }}
		    comment-pr: ${{ inputs.comment-on-pr }}
```

### Why is rolling back not an option for us?

I tested the previous release v13.6.1 to see if rolling back is an option for us, but there we are hit by another issue, that seems to be fixed in v13.7.0:

**(Left v13.6.1 Right v13.7.0)**

<img width="2126" height="506" alt="Image" src="https://github.com/user-attachments/assets/558848cf-a48e-488f-a8a3-3864b9ba74f4" />

```txt
jq: error (at <stdin>:1): string ("10-tv2-play") and null (null) cannot have their containment checked
Error: Process completed with exit code 5.
```

A bit complex but our use case seems to be recognized as a matrix, as we have a workflow (without a matrix) using a reusable workflow (with a matrix). It then proceeds to fail on v13.6.1 because it is unable to handle null values in the related code (see above). We do not encounter this issue on v13.7.0, so the latests changes seems to fix this problem for our use case as well.

This leaves us with the problem @birjj describes above, for us to proceed with getting our solution to a working state.

--- a/action.yml
+++ b/action.yml
@@ -432,7 +432,6 @@ runs:
 
         # Set flags for creating PR comment and tagging actor.
         create_comment=""
-        delete_comment=""
         tag_actor=""
         if [[ "$INPUTS_TAG_ACTOR" == "always" || "$INPUTS_TAG_ACTOR" == "true" ]]; then
           tag_actor="@"
@@ -477,40 +476,45 @@ runs:
         # Clean up files.
         rm --force tf.command.txt tf.console.txt tf.diff.txt
 
-        # Determine if a PR comment should be created.
-        if [[ "$INPUTS_COMMENT_PR" == "always" && "${{ steps.identifier.outputs.pr }}" -ne 0 ]]; then
-          create_comment="true"
-        elif [[ ("$INPUTS_COMMENT_PR" == "on-diff" || "$INPUTS_COMMENT_PR" == "on-change") && "$exitcode" -ne 0 && "${{ steps.identifier.outputs.pr }}" -ne 0 ]]; then
-          create_comment="true"
-        elif [[ ("$INPUTS_COMMENT_PR" == "on-diff" || "$INPUTS_COMMENT_PR" == "on-change") && "$exitcode" -eq 0 && "${{ steps.identifier.outputs.pr }}" -ne 0 ]]; then
-          delete_comment="true"
+        # The remainder of the code relates to commenting on the PR.
+        # Exit early if we don't have a relevant PR.
+        if [[ "${{ steps.identifier.outputs.pr }}" -eq 0 ]]; then
+          exit 0
         fi
 
-        if [[ "$create_comment" == "true" || "$delete_comment" == "true" ]]; then
-          # Check if the PR contains a bot comment with the same identifier.
-          # We do this here to avoid attempting to delete/create a comment in a non-existant PR.
-          list_comments=$(gh api /repos/${{ github.repository }}/issues/${{ steps.identifier.outputs.pr }}/comments --header "$GH_API" --method GET --paginate)
-          bot_comment=$(echo "$list_comments" | jq --raw-output --arg identifier "${{ steps.identifier.outputs.name }}" '.[] | select(.user.type == "Bot") | select(.body | contains($identifier)) | .id' | tail -n 1)
+        # Check if the PR contains a bot comment with the same identifier.
+        list_comments=$(gh api /repos/${{ github.repository }}/issues/${{ steps.identifier.outputs.pr }}/comments --header "$GH_API" --method GET --paginate)
+        bot_comment=$(echo "$list_comments" | jq --raw-output --arg identifier "${{ steps.identifier.outputs.name }}" '.[] | select(.user.type == "Bot") | select(.body | contains($identifier)) | .id' | tail -n 1)
 
-          # Either delete, update or create a new comment.
-          if [[ "$delete_comment" == "true" && -n "$bot_comment" ]]; then
+        # Determine if a PR comment should be created.
+        if [[ "$INPUTS_COMMENT_PR" == "always" ]]; then
+          create_comment="true"
+        elif [[ ("$INPUTS_COMMENT_PR" == "on-diff" || "$INPUTS_COMMENT_PR" == "on-change") && "$exitcode" -ne 0 ]]; then
+          create_comment="true"
+        elif [[ ("$INPUTS_COMMENT_PR" == "on-diff" || "$INPUTS_COMMENT_PR" == "on-change") && "$exitcode" -eq 0 && -n "$bot_comment" ]]; then
+          gh api /repos/${{ github.repository }}/issues/comments/${bot_comment} --header "$GH_API" --method DELETE
+          exit 0
+        fi
+        # Exit early if no comment is needed.
+        if [[ "$create_comment" != "true" ]]; then
+          exit 0
+        fi
+        # Create PR comment based on configuration and existing comment.
+        if [[ -n "$bot_comment" ]]; then
+          if [[ "$INPUTS_COMMENT_METHOD" == "update" ]]; then
+            # Update existing comment.
+            pr_comment=$(gh api /repos/${{ github.repository }}/issues/comments/${bot_comment} --header "$GH_API" --method PATCH --field "body=${body}")
+            echo "comment_id=$(echo "$pr_comment" | jq --raw-output '.id')" >> "$GITHUB_OUTPUT"
+          elif [[ "$INPUTS_COMMENT_METHOD" == "recreate" ]]; then
+            # Delete previous comment before posting a new one.
             gh api /repos/${{ github.repository }}/issues/comments/${bot_comment} --header "$GH_API" --method DELETE
-          elif [[ "$create_comment" == "true" && -n "$bot_comment" ]]; then
-            if [[ "$INPUTS_COMMENT_METHOD" == "update" ]]; then
-              # Update existing comment.
-              pr_comment=$(gh api /repos/${{ github.repository }}/issues/comments/${bot_comment} --header "$GH_API" --method PATCH --field "body=${body}")
-              echo "comment_id=$(echo "$pr_comment" | jq --raw-output '.id')" >> "$GITHUB_OUTPUT"
-            elif [[ "$INPUTS_COMMENT_METHOD" == "recreate" ]]; then
-              # Delete previous comment before posting a new one.
-              gh api /repos/${{ github.repository }}/issues/comments/${bot_comment} --header "$GH_API" --method DELETE
-              pr_comment=$(gh api /repos/${{ github.repository }}/issues/${{ steps.identifier.outputs.pr }}/comments --header "$GH_API" --method POST --field "body=${body}")
-              echo "comment_id=$(echo "$pr_comment" | jq --raw-output '.id')" >> "$GITHUB_OUTPUT"
-            fi
-          elif [[ "$create_comment" == "true" ]]; then
-            # Post new comment.
             pr_comment=$(gh api /repos/${{ github.repository }}/issues/${{ steps.identifier.outputs.pr }}/comments --header "$GH_API" --method POST --field "body=${body}")
             echo "comment_id=$(echo "$pr_comment" | jq --raw-output '.id')" >> "$GITHUB_OUTPUT"
           fi
+        else
+          # Post new comment.
+          pr_comment=$(gh api /repos/${{ github.repository }}/issues/${{ steps.identifier.outputs.pr }}/comments --header "$GH_API" --method POST --field "body=${body}")
+          echo "comment_id=$(echo "$pr_comment" | jq --raw-output '.id')" >> "$GITHUB_OUTPUT"
         fi
 
 outputs:

